src/pages/MatchingPage/index.js

```diff
@@ -1,25 +1,59 @@
 import React, { useState, useEffect } from 'react';
 
 import Card from '../../components/Card';
-import {matching} from '../../data/index.json';
 
-const leftCards = randElements(
-  matching.map(({ word }, index) => ({ id: index, card: word })),
-);
-const rightCards = randElements(
-  matching.map(({ definition }, index) => ({ id: index, card: definition })),
-);
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { fetchMatchingGameData } from '../../actions/matching';
+
+const mapStateToProps = (props) => ({
+  matching: props.matching.matching,
+});
+
+const mapDispatchToProps = (dispatch) =>
+  bindActionCreators(
+    {
+      fetchMatchingGameData: fetchMatchingGameData,
+    },
+    dispatch,
+  );
 
 function randElements(cards) {
   let randCards = [...cards];
   randCards.sort(() => Math.random() - 0.5);
   return randCards;
 }
+const MatchingCard = (props) => {
+  const { matching = [] } = props;
 
-function MatchingCard() {
   const [selectedCard, setSelectedCard] = useState({ left: null, right: null });
   const [guessedCards, setGuesssedCards] = useState([]);
 
+  const [leftCards, setLeftCards] = useState([]);
+  const [rightCards, setRightCards] = useState([]);
+
+  useEffect(() => {
+    setLeftCards(
+      randElements(
+        matching.map(({ word }, index) => ({ id: index, card: word })),
+      ),
+    );
+    setRightCards(
+      randElements(
+        matching.map(({ definition }, index) => ({
+          id: index,
+          card: definition,
+        })),
+      ),
+    );
+  }, [matching]);
+
+  useEffect(() => {
+    props.fetchMatchingGameData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do you need to disable eslint here?

Reply:
using 'disable eslint' i remove this warning
![Screenshot_5](https://user-images.githubusercontent.com/17178553/93916570-6c833c00-fd12-11ea-8cff-04bb91c773ea.png)

Reply:
Line 55:23:  Assignments to the 'timerChoose' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps

Reply:
ook, this problem can be fixed without disabling eslint.

Could you try to move

https://github.com/iviaks/english-games/blob/e6f64b230e6290465ddf637d77e534be1d261e42/src/pages/MatchingPage/index.js#L47 

onto these lines

https://github.com/iviaks/english-games/blob/e6f64b230e6290465ddf637d77e534be1d261e42/src/pages/MatchingPage/index.js#L11-L12

so, the result will be

```diff
+let timerChoose;
+
function randElements(cards) {
  let randCards = [...cards];
...

  useEffect(() => {
    props.fetchFirebaseData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

- let timerChoose;
```

I hope, this moving will remove the warning.